client/src/components/ProfilePage/Location/index.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import Inputs from '../../common/Inputs';
+import Button from '../../common/Button';
+import './style.css';
+
+class Location extends Component {
+  state = {
+    validation: false,
+    found: false,
+  };
+
+  onChange = (e) => {
+    const { name, value } = e.target;
+    this.setState({ [name]: value, validation: false, found: false });
+  };
+
+  onClick=() => {
+    const { City, State } = this.state;
+    if (City && City.trim() && State && State.trim()) {
+      const apiKey = '219fa7fc98c345d1b38193fa9201a96e';
+      axios.get(`https://api.opencagedata.com/geocode/v1/json?q=${State},${City}&key=${apiKey}`).then((res) => {
+        if (res.data.total_results !== 0) {
+          const locationValues = res.data.results[0].geometry;
+          const { lat, lng } = locationValues;
+          const data = { State, lat, lng };
+          axios.post('api/v1/geo-code', data).then(
+            this.setState({ message: 'Successfully added ', validation: false, found: false }),
+          ).catch((error) => {
+            const { status } = error.response;
+            if (status === 404) {
+              this.setState({ message: 'Page Not Found' });
+            }
+          });
+        } else {
+          this.setState({ messageNotFound: ' Not Found', found: true });
```

Comment:
@ali-7 Does setting `found: true` here mean that the button will be hidden?  That seems like it would prevent the user from fixing their input and re-submitting, in this error case.

Reply:
@ali-7  it will be true only when there's error, and the message will display for the user, the user will be ok to re-submit there inputs